biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java

```diff
@@ -594,29 +594,34 @@ private boolean isMaster(InfoRepository repo, String bsn, Version v) throws Exce
 	}
 
 	private RepositoryPlugin getReleaseRepo() {
-		String repoName = getProperty(Constants.RELEASEREPO);
 
-		List<RepositoryPlugin> repos = getPlugins(RepositoryPlugin.class);
-		for (RepositoryPlugin r : repos) {
-			if (r.canWrite()) {
-				if (repoName == null || r.getName()
-					.equals(repoName)) {
-					return r;
-				}
+		String repoNames = getProperty(Constants.RELEASEREPO);
+
+		if (repoNames == null) {
+			error("Could not find a release reposiotry (-releaserepo is not set)");
```

Comment:
There is a misspelling here. Also, we agreed that missing `-releaserepo` was a warning and not an error. And if `-releaserepo` was not set, we still release to the first writable repo for backwards compatibility.